jeff65/gold/lexer.py

```diff
@@ -95,7 +96,10 @@
 }
 
 # non-whitespace characters which can end words.
-specials = '()[]{}:.,"\\'
+specials = '()[]{}:;.,"\\@'
+
+# non-whitespace characters which can end sprinkles.
+sprinkle_specials = ','
```

Comment:
TODO: this is a hack and doesn't even fix the nested brackets problem

Reply:
it does prevent `],` from being read as a single token, but so would moving `[` and `]` to the `nestable_delimiters` list. that still won't help disambiguate between the comment close and two `]` tokens, though.